deployment/data-streams/changeset/common.go

```diff
@@ -0,0 +1,74 @@
+package changeset
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+const (
+	defaultJobSpecsTimeout = 120 * time.Second
+)
+
+// TODO: This file needs to be renamed and rethought. It's a good first step, though.
+
+func findConfiguratorAddressByDON(addresses map[string]deployment.TypeAndVersion, donID uint64) (string, error) {
+	for address, contract := range addresses {
+		if contract.Type == "Configurator" && contract.Labels.Contains(fmt.Sprintf("don-%d", donID)) {
```

Comment:
Can we have multiple configurator versions at the same time?

Reply:
Hm, that's a good point. Looking at [deploy_configurator](https://github.com/smartcontractkit/chainlink/blob/28c39bf1a02610a07fd38d055efef91ee3a37c37/deployment/data-streams/changeset/configurator/v0_5_0/deploy_configurator.go#L83-L83) I don't see anything removing any configurators for the DON. So, yes, we can have multiple of them. This probably means that we need the configurator address as part of the inputs of this changeset. We do have that in `dons.json`.

Reply:
It makes no sense to try to guess the configurator address when we can simply pass it in the configuration inputs.
Good callout!